backend/tournesol/views/exports.py

```diff
@@ -35,6 +35,43 @@ def write_comparisons_file(request, write_target):
     )
 
 
+def write_public_comparisons_file(request, write_target):
+    """
+    Writes all public comparisons data as a CSV file to write_target which can be
+    among other options an HttpResponse or a StringIO
+    """
+    fieldnames = ['public_username', 'video_a', 'video_b', 'criteria', 'weight', 'score']
+    writer = csv.DictWriter(write_target, fieldnames=fieldnames)
+    writer.writeheader()
+    public_data = ContributorRating.objects.filter(is_public=True).select_related("user", "video")
+    serialized_comparisons = []
+    public_videos = set()
+    users = set()
+    for public_video in public_data:
+        public_videos.add((public_video.user, public_video.video))
+        users.add(public_video.user)
+    for user in users:
+        comparisons = Comparison.objects.filter(user=user)\
```

Comment:
This is going to be a performance bottleneck because it does a database query for each user (that means a few thousands of database queries right now).

```Python
comparisons = Comparison.objects.all().select_related("video_1", "video_2").prefetch_related("criteria_scores")

public_comparisons = [comparison for comparison in comparisons if (
    (comparison.user, comparison.video_1) in public_videos and (comparison.user, comparison.video_2) in public_videos
)]
```